core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -1738,6 +1738,20 @@ private void checkExponentialCnf(int n) {
         "OR(null, <>(0, ?0.int0))",
         "<>(0, ?0.int0)",
         "true");
+
+    // (a=1 or a=2 or (arr[1]>4 and arr[1]<3 and a=3)) => a=1 or a=2
+    final RelDataType intArrayType = typeFactory.createArrayType(intType, -1);
+    final RexInputRef ref0 = rexBuilder.makeInputRef(intType, 0);
+    final RexInputRef ref3 = rexBuilder.makeInputRef(intArrayType, 3);
+    final RexCall itm1 = (RexCall) rexBuilder.makeCall(intType, SqlStdOperatorTable.ITEM,
+        ImmutableList.of(ref3, literal1));
+    simplify = this.simplify.withParanoid(false);
+    checkSimplifyFilter(
+        or(eq(ref0, literal1), eq(ref0, literal2),
```

Comment:
I think we'd better create a new method to test it. This method is redundant now.

Reply:
I think this is exactly place where everybody places such tests. Also I'm using here some general local vars declared earlier.

Reply:
That's what I am worried about. This method contains too many tests.

Reply:
How did you measure when it's too many tests in the method? As you may notice the practice of making tests as small as possible was already sacrificed long time ago. And refactoring of tests is out of scope of this PR.

Reply:
@ihuzenko , one test case per method is OK.

>And refactoring of tests is out of scope of this PR.

You add a single test case (which is great), and it is good that the test is almost isolated.

It is absolutely not clear why do you add the test to an existing test method. Is is complicated to add a new method?

Note: "long test methods" like `testSimplifyFilter` and `testSimplifyOrTerms` was created only because it was complicated to initialize `RexLiteral` and similar objects for the test.
Nowadays, you can use `literal(1)`, and you do not need `literal1` variable.
In the same way, you can use `vInt(0)` and `vInt(3)` which eliminates the need for `ref0` and `ref3`.

Apparently `RexProgramBuilderBase` misses the DSL for `SqlStdOperatorTable.ITEM` so you can leave it in your test method or add `item` DSL to `RexProgramBuilderBase`.

Could you please consider moving the added test case to its own method and use `vInt(...)` for variables?

Reply:
Test was added to existing method because it looks like it's common practice in the class. Of course it's easy to use new test method and I'll do it. What I want to say is that it doesn't change things much, even worse, the class won't have any consistency and will continue to confuse new contributors.